web-portal/components/SubmissionForm.vue

```diff
@@ -517,14 +523,52 @@
       },
       selectVenue: function (venue) {
         this.calendar_event.venue_id = venue.id
+        this.doTimeAndLocationExistingEventDetection()
       },
       newVenue: function (venue) {
         this.calendar_event.venue_id = venue.id
         this.$refs.venuePicker.handleNewVenue(venue)
+        this.doTimeAndLocationExistingEventDetection()
       },
+      doTimeAndLocationExistingEventDetection: function() {
+        this.duplicateEventsByStartTime = null
+        const venueId = this.calendar_event.venue_id
+        const dateTimes = this.calendar_event.date_times
+
+        if (!this.isAdmin) {
+          // for now, we will only expose duplicate detection to ourselves. We are using admin kind of like a feature flag
+          return
+        } else if (venueId === undefined || venueId === null) {
+          // we need a venue to do the check
+          return
+        } else if (!Array.isArray(dateTimes) || dateTimes.length === 0) {
+          // we need at least one start time to do the check
+          return
+        }
 
+        const duplicateDetectionPayload = {
+          timeAndLocations: dateTimes.map(({ start_time }) => ({
+            venueId: venueId,
+            startTime: start_time
+          }))
+        }
+
+        this.$apiService.post('/events/detect-existing/by-time-and-location', duplicateDetectionPayload)
+          .then((resp) => {
+            this.duplicateEventsByStartTime = resp.data || null
+          })
+          .catch((err) => {
+            console.error('Error performing duplicate detection on start times: ', err)
+          })
+      },
+      shouldShowExistingEventDetectionByStartTime: function () {
+        return this.duplicateEventsByStartTime !== null &&
+          this.duplicateEventsByStartTime !== undefined &&
+          this.duplicateEventsByStartTime.isLikelyExisting
+      },
```

Comment:
This could be a computed property instead of a method.

Reply:
1aad33ca29cb31e60a0b40ab558690885e142d78

Reply:
1aad33ca29cb31e60a0b40ab558690885e142d78